README.md

```diff
@@ -1 +1,1148 @@
-# balena-cli-masterclass
\ No newline at end of file
+balenaCLI Masterclass
+=====================
+
+**Masterclass Type:** Core
+
+# Introduction
+
+The balena Command Line Interface (balenaCLI) utility consists of a number
+of commands that allow a user to develop, deploy and manage balena applications
+and devices.
+
+Everything that can be achieved via the balenaCloud dashboard can also be
+achieved via the balenaCLI.
+
+In this masterclass, you will learn how to:
+
+* Login to your account
+* Push application code to a balena Application
+* Deploy locally built code to a balena Application
+* SSH into a balena device
+* Push and build applications on a device on the local network for fast
+    development and prototyping
+* Use private Docker registries and modules from a balena device
+
+# Hardware and Software Requirements
+
+It is assumed that the reader has access to the following:
+
+* A [balenaFin 1.1](https://store.balena.io/collections/developer-kit/products/balenafin-v1-1-0-developer-kit)
+* A suitable text editor for developing code on your development platform (eg.
+    [Visual Code](https://code.visualstudio.com/))
+* A suitable shell environment for command execution (such as `bash`)
+* A [balenaCloud](https://www.balena.io/) account
+* A familiarity with [Dockerfiles](https://docs.docker.com/engine/reference/builder/)
+* A [NodeJS](https://nodejs.org/en/) installation, including [NPM](https://www.npmjs.com/get-npm)
+
+# Exercises
+
+All of the following exercises assume that you are running the balenaCLI from
+a suitable Unix based shell. The exercises include commands which can be run
+in such a shell, and are represented by a line prefixed with `$`. Information
```

Comment:
Convention is typically "commands as your user `$`, commands as root `#`. I _think_ at least the `npm --global` requires superuser permissions.

Reply:
This depends on you node/NPM installation. I use `nvm` where `global` is restricted to the current user and version, and I think that's actually now worth commenting on in the text.

Reply:
Done!